vhpc_toolkit/operations.py

```diff
@@ -1989,3 +1990,109 @@ def _destroy_cluster_dvs(self, switch_cfgs):
+
+    def get_vm_config_cli(self) -> None:
+        vm_cfgs = self._extract_file(self.cfg)
+        for vm_cfg in vm_cfgs:
+            self._print_vm_config(vm_cfg["vm"])
+
+    def _print_vm_config(self, vm_name: str) -> None:
+        """
+        This function prints the performance related settings of the vm using the vm name provided
+
+        Args:
+            vm_name: Name of the VM for which you want the performance related settings printed
+        Returns:
+            None
+        """
+        vm_object: vim.VirtualMachine = self.objs.get_vm(vm_name)
+        vm = GetVM(vm_object)
+
+        existing_pci_ids = vm.existing_pci_ids()
+        # Get PCI devices attached to the current VM
+        attached_direct_passthru_devices = []
+        for passthru_device in vm.avail_pci_info():
+            if passthru_device[2] in existing_pci_ids:
+                device_name, vendor_name, device_id, system_id = passthru_device
+                attached_direct_passthru_devices.append(
+                    {
+                        "Device Name": f"{vendor_name} - {device_name}",
+                        "Device ID": device_id,
+                    }
+                )
+
+        # Get SRIOV devices to the current VM
+        attached_sriov_devices = []
+        existing_sriov_ids = vm.existing_sriov_ids()
+
+        for sriov_device in vm.avail_sriov_info():
+            if sriov_device[4] in existing_sriov_ids:
+                (
+                    pnic,
+                    virtual_function,
+                    device_name,
+                    vendor_name,
+                    device_id,
+                    system_id,
+                ) = sriov_device
+                attached_sriov_devices.append(
+                    {
+                        "PNIC": pnic,
+                        "Virtual Function": virtual_function,
+                        "Device Name": f"{vendor_name} - {device_name}",
+                        "Device ID": device_id,
+                    }
+                )
+
+        attached_pvrmda_devices = []
+
+        for network_object in vm.device_objs_all():
+            if isinstance(
+                network_object, vim.vm.device.VirtualSriovEthernetCard
+            ) and hasattr(network_object, "sriovBacking"):
+                for attached_sriov_device in attached_sriov_devices:
+                    if (
+                        attached_sriov_device["Device ID"]
+                        == network_object.sriovBacking.physicalFunctionBacking.id
+                    ):
+                        attached_sriov_device.update(
+                            {"Label": network_object.deviceInfo.label}
+                        )
+            if isinstance(network_object, vim.vm.device.VirtualVmxnet3Vrdma):
+                attached_pvrmda_devices.append(
+                    {
+                        "Device ID": network_object.macAddress,
```

Comment:
Device ID should be really MAC address, but it's not useful, so let's remove this line